pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts

```diff
@@ -10,14 +10,61 @@ const stringify = (map: LabelSet): string => {
 
 const sortBlocksInRows = (blocks: Block[]): BlocksPool => {
   const pool: BlocksPool = {};
+  const minTimeCount = new Map();
+  const maxTimeCount = new Map();
+  const maxToMinTime = new Map();
 
   blocks
-    .sort((a, b) => a.thanos.downsample.resolution - b.thanos.downsample.resolution)
-    .forEach(b => {
-      if (!pool[`${b.compaction.level}-${b.thanos.downsample.resolution}`])
-        pool[`${b.compaction.level}-${b.thanos.downsample.resolution}`] = [];
+    .sort((a, b) => {
+      if (a.minTime - b.minTime === 0) return a.thanos.downsample.resolution - b.thanos.downsample.resolution;
+      else return a.minTime - b.minTime;
+    })
+    .forEach((b, index, blocks) => {
+      const currMin = `${b.minTime}`;
+      const currMax = `${b.maxTime}`;
+
+      if (!maxToMinTime.has(currMax)) maxToMinTime.set(currMax, currMin);
+
+      if (index === 0) {
+        minTimeCount.set(currMin, 0);
+        maxTimeCount.set(currMax, 0);
+      } else {
+        const preMin = `${blocks[index - 1].minTime}`;
+        const preMax = `${blocks[index - 1].maxTime}`;
+        let nearMax: any;
+
+        maxTimeCount.forEach((v, k) => {
+          if (k <= currMin) nearMax = k;
+        });
+
+        if (currMin < preMax) {
```

Comment:
Nice work! 

I wonder if it has to be so many if/elses =D

Good for a start I guess (: Not a blocker - just bit complex, code maybe there is a way simplify

Reply:
It's true that it is a bit complex or we can say that is the brute force approach for the problem but I tried and didn't find any optimal approach for it. If anyone could suggest any better/optimized approach, I would be happy to implement it otherwise I think we can merge it for now after making a separate function for the if-else ladder as suggested by @prmsrswt. Thanks!

Reply:
Tip: Recursion of function helpers (:

Reply:
If we do `isOverlapping` function would be nice (: